### ISSUE TITLE
ganeti-watcher wrongly restarts ganeti-rapi that requires authentication

### ISSUE THREAD

**Issue by Ganeti-Issues-Migrator** [ganeti/ganeti#808]
Originally reported of Google Code with ID 752.

```
<b>What steps will reproduce the problem?</b>
1. Set '--require-authentication' option to ganeti-rapi
2. Wait for ganeti-watcher to run

<b>What is the expected output? What do you see instead?</b>

ganeti-watcher will always restart ganeti-rapi daemon:

2014-03-07 12:45:01,965: ganeti-watcher pid=17267 WARNING RAPI error: 401 Unauthorized: No permission -- see authorization schemes
2014-03-07 12:45:01,967: ganeti-watcher pid=17267 WARNING Couldn't get answer from remote API, restaring daemon
2014-03-07 12:45:01,968: ganeti-watcher pid=17267 INFO RunCmd /usr/lib/ganeti/daemon-util stop ganeti-rapi
2014-03-07 12:45:02,301: ganeti-watcher pid=17267 INFO RunCmd /usr/lib/ganeti/daemon-util check-and-start ganeti-rapi
2014-03-07 12:45:03,683: ganeti-watcher pid=17267 WARNING RAPI error: 401 Unauthorized: No permission -- see authorization schemes
2014-03-07 12:45:03,684: ganeti-watcher pid=17267 CRITICAL RAPI is not responding

ganeti-watcher connects to RAPI without user credentials, and interprets the authentication issue
as RAPI daemon not responding. Thus it will restart the daemon.


```
Originally added on 2014-03-07 10:54:22 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
Hi, could you please provide the output of "gnt-cluster version"?
```
Originally added on 2014-03-07 12:32:19 +0000 UTC.

Changed State: NeedFeedback

**Comment by Ganeti-Issues-Migrator** [comment]
```
Hi Michele,

$ gnt-cluster version
Software version: 2.8.2
Internode protocol: 2080000
Configuration format: 2080000
OS api version: 20
Export interface: 0
VCS version: v2.8.0-27-g68367e2

This issue was introduced with '--require-authentication' option
in Ganeti 2.8. I believe that it remains until current master. 
```
Originally added on 2014-03-07 12:41:00 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
Ok, thanks.
```
Originally added on 2014-03-07 12:47:22 +0000 UTC.

Changed State: New

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2014-03-07 13:40:28 +0000 UTC.

Added Labels: Type-Defect Priority-High
Changed State: Accepted
Added to Milestone: Release2.8

**Comment by Ganeti-Issues-Migrator** [comment]
```
commit 6177890bde9d40b17062c4e896950f1e1f6df361
Author: Michele Tartara <mtartara@google.com>
Date:   Fri Mar 7 13:59:49 2014 +0000

    Improve RAPI detection of the watcher
    
    If the watcher is not allowed to access RAPI, it doesn't mean that it is dead
    and needs to be restarted.
    
    Fixes Issue 752.
    
    Signed-off-by: Michele Tartara <mtartara@google.com>
    Reviewed-by: Hrvoje Ribicic <riba@google.com>

```
Originally added on 2014-03-07 15:36:10 +0000 UTC.

Mentioned Issue with GoogleCode ID 752 has been migrated to GitHub with ID #808.
Changed State: Fixed

**Comment by Ganeti-Issues-Migrator** [comment]
```
Released as part of 2.10.2.
```
Originally added on 2014-03-24 12:18:46 +0000 UTC.

Changed State: Released
